src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java

```diff
@@ -57,4 +57,12 @@ public WorkflowStep createStep(String type) {
         }
         throw new FlowFrameworkException("Workflow step type [" + type + "] is not implemented.", RestStatus.NOT_IMPLEMENTED);
     }
+
+    /**
+     * Gets the step map
+     * @return the step map
+     */
+    public Map<String, WorkflowStep> getStepMap() {
+        return this.stepMap;
```

Comment:
Sorry I missed this in earlier review. If we want to make this public we need a `copyOf()` to prevent the caller from modifying the map.